src/main/java/org/reaktivity/nukleus/http2/internal/stream/Http2ServerFactory.java

```diff
@@ -2925,10 +2985,15 @@ private void doResponseResetIfNecessary(
             {
                 correlations.remove(responseId);
 
-                if (!Http2State.replyClosed(state))
+                if (Http2State.replyOpened(state) && !Http2State.replyClosed(state))
                 {
                     doResponseReset(traceId, authorization);
                 }
+                else if (streams.containsKey(streamId))
+                {
+                    setResponseClosed();
+                    removeStream();
+                }
```

Comment:
This explicit call to `removeStream()` doesn't seem right as it should be happening automatically if the state is correct for both `request` and `response`.
What goes wrong here if this `else` block is removed?
Also, it is not wrong to send the `RESET` proactively when the correlation is removed, since a subsequent `BEGIN` reply will get `RESET` anyway when the correlation id is not found, right?